vpr/src/route/partition_tree.cpp

```diff
@@ -40,50 +53,58 @@ std::unique_ptr<PartitionTreeNode> PartitionTree::build_helper(const Netlist<>&
         /* Inclusive start and end coords of the bbox relative to x1. Clamp to [x1, x2]. */
         int x_start = std::max(x1, bb.xmin) - x1;
         int x_end = std::min(bb.xmax, x2) - x1;
-        /* Fill in the lookups assuming a cutline at x + 0.5. */
-        for (int x = x_start; x < W - 1; x++) {
+        /* Fill in the lookups assuming a cutline at x + 0.5.
+         * This means total_before includes the max coord of the bbox but
+         * total_after does not include the min coord. */
+        for (int x = x_end; x < W - 1; x++) {
             x_total_before[x] += fanouts;
         }
-        for (int x = 0; x < x_end; x++) {
+        for (int x = 0; x < x_start; x++) {
             x_total_after[x] += fanouts;
         }
+        for (int x = x_start; x < x_end; x++){
+            x_total_on[x] += fanouts;
+        }
         int y_start = std::max(y1, bb.ymin) - y1;
         int y_end = std::min(bb.ymax, y2) - y1;
-        for (int y = y_start; y < H - 1; y++) {
+        for (int y = y_end; y < H - 1; y++) {
             y_total_before[y] += fanouts;
         }
-        for (int y = 0; y < y_end; y++) {
+        for (int y = 0; y < y_start; y++) {
             y_total_after[y] += fanouts;
         }
+        for (int y = y_start; y < y_end; y++){
+            y_total_on[y] += fanouts;
+        }
     }
 
-    int best_score = std::numeric_limits<int>::max();
+    uint64_t best_score = std::numeric_limits<uint64_t>::max();
     float best_pos = std::numeric_limits<double>::quiet_NaN();
     Axis best_axis = Axis::X;
 
-    int max_x_before = x_total_before[W - 2];
-    int max_x_after = x_total_after[0];
     for (int x = 0; x < W - 1; x++) {
```

Comment:
Not sure where W and H come from.  Width and height of the bounding box of this subtree.  They're also pretty short names -- would longer names be useful?